sky/spot/dashboard/dashboard.py

```diff
@@ -0,0 +1,66 @@
+"""Dashboard for spot jobs based on Flask."""
+import datetime
+import pathlib
+import re
+import yaml
+
+import flask
+
+import sky
+from sky import spot
+
+app = flask.Flask(__name__)
+
+
+def _is_running_on_spot_controller() -> bool:
+    """Am I running on spot controller?
+
+    Loads ~/.sky/sky_ray.yml and check cluster_name.
+    """
+    if pathlib.Path('~/.sky/sky_ray.yml').expanduser().exists():
+        config = yaml.safe_load(
+            pathlib.Path('~/.sky/sky_ray.yml').expanduser().read_text())
+        return config.get('cluster_name', '').startswith('sky-spot-controller-')
+    return False
+
+
+@app.route('/')
+def home():
+    if not _is_running_on_spot_controller():
+        # Experimental: run on laptop (refresh is very slow).
+        all_spot_jobs = sky.spot_queue(refresh=True, skip_finished=False)
+    else:
+        job_table = spot.dump_spot_job_queue()
+        all_spot_jobs = spot.load_spot_job_queue(job_table)
+
+    timestamp = datetime.datetime.utcnow()
+    rows = spot.format_job_table(all_spot_jobs, show_all=True, return_rows=True)
+
+    # FIXME(zongheng): make the job table/queue funcs return structured info so
+    # that we don't have to do things like row[-5] below.
+    columns = [
+        'ID', 'Task', 'Name', 'Resources', 'Submitted', 'Total Duration',
```

Comment:
nit: Is it possible to make `Resources`, `Submitted` and `Started` columns a bit wider, to avoid the new lines?

Reply:
Played around with it by adding 
```
white-space: nowrap;
```
to some columns. However, it doesn't look too good when the window is occupying half of the monitor.

How about we do the UI finetuning later preferably with some UI folks? We may also want to drop some of these columns.

Reply:
That sounds good to me! Let's have the current version in first.